server/data/data-jmap/src/test/java/org/apache/james/jmap/api/projections/MessageFastViewProjectionHealthCheckTest.java

```diff
@@ -0,0 +1,186 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.api.projections;
+
+import static org.apache.james.jmap.api.projections.MessageFastViewProjection.METRIC_RETRIEVE_HIT_COUNT;
+import static org.apache.james.jmap.api.projections.MessageFastViewProjection.METRIC_RETRIEVE_MISS_COUNT;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.stream.IntStream;
+
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.metrics.api.Metric;
+import org.apache.james.metrics.tests.RecordingMetricFactory;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class MessageFastViewProjectionHealthCheckTest {
+
+    private static final ComponentName COMPONENT_NAME = new ComponentName("MessageFastViewProjection");
+
+    private MessageFastViewProjectionHealthCheck testee;
+    private Metric hitMetric;
+    private Metric missMetric;
+
+    @BeforeEach
+    void setUp() {
+        RecordingMetricFactory metricFactory = new RecordingMetricFactory();
+        testee = new MessageFastViewProjectionHealthCheck(metricFactory);
+
+        hitMetric = metricFactory.generate(METRIC_RETRIEVE_HIT_COUNT);
+        missMetric = metricFactory.generate(METRIC_RETRIEVE_MISS_COUNT);
+    }
+
+    @Test
+    void componentNameShouldReturnTheRightValue() {
+        assertThat(testee.componentName())
+            .isEqualTo(COMPONENT_NAME);
+    }
+
+    @Nested
+    class WithTenPercentMaximumOfMissCount {
+
+        @Test
+        void checkShouldReturnHealthyWhenNoRetrieveCalled() {
+            assertThat(testee.check())
+                .isEqualTo(Result.healthy(COMPONENT_NAME));
+        }
+
+        @Test
+        void checkShouldReturnDegradedWhenNoHitButSomeMiss() {
+            missMetric.increment();
+            missMetric.increment();
+
+            assertThat(testee.check())
+                .isEqualTo(Result.degraded(COMPONENT_NAME, "retrieveMissCount percentage 100.0% (2/2) is higher than the threshold 10.0%"));
+        }
+
+        @Test
+        void checkShouldReturnHealthyWhenMissCountPercentageIsGreaterThanThreshold() {
```

Comment:
Are you sure about the name? It is not when miss count percentage is lower than threshold that it's healthy?

checkShouldReturnHealthyWhenMissCountPercentageIsLowerThanThreshold ?